core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/localscan/LocalCsvFileScanOpDesc.java

```diff
@@ -86,17 +82,102 @@ public Schema sourceSchema() {
+        java.util.List<AttributeType> attributeType = IntStream
+                .range(0, headerLine.split(delimiter).length).mapToObj(i->AttributeType.INTEGER).collect(Collectors.toList());
+
+
+        try(BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            if (header != null && header)
+                reader.readLine();
+            int i = 0;
+            while((line = reader.readLine())!=null && i<=100) {
+                String[] records = line.split((delimiter));
+                attributeType = inferline(attributeType, records);
+                i++;
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        java.util.List<AttributeType> finalAttributeType = attributeType;
+        System.out.println("a->"+finalAttributeType);
+//        System.out.println(Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
+//                mapToObj(i -> new Attribute(l.get(i), finalAttributeType.get(i)))
+//                .collect(Collectors.toList())).build());
+
         if (delimiter == null) {
             return null;
         }
         if (header != null && header) {
-            return Schema.newBuilder().add(Arrays.stream(headerLine.split(delimiter)).map(c -> c.trim())
-                    .map(c -> new Attribute(c, AttributeType.STRING)).collect(Collectors.toList())).build();
+
+
+            return Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
+                    mapToObj(i -> new Attribute(headerLine.split(delimiter)[i], finalAttributeType.get(i)))
+                    .collect(Collectors.toList())).build();
+//            return Schema.newBuilder().add(Arrays.stream(headerLine.split(delimiter)).map(c -> c.trim())
+//                    .map(c -> new Attribute(c, AttributeType.STRING)).collect(Collectors.toList())).build();
         } else {
             return Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
-                    mapToObj(i -> new Attribute("column" + i, AttributeType.STRING))
+                    mapToObj(i -> new Attribute("column" + i, finalAttributeType.get(i)))
                     .collect(Collectors.toList())).build();
         }
     }
+    private java.util.List<AttributeType> inferline (java.util.List<AttributeType> attributeType, String[] line) {
+        return IntStream.range(0, line.length).
+                mapToObj(i->inferChar(attributeType.get(i),line[i])).collect(Collectors.toList());
+    }
+
+    private AttributeType inferChar (AttributeType attributeType, String c) {
+        if (attributeType.getName().equals("string")) {
+            return AttributeType.STRING;
+        } else if (attributeType.getName().equals("boolean")) {
+            try {
+                Integer.parseInt(c);
```

Comment:
boolean and int are different. 
boolean values could be 1, 0, true, false; would you infer a value of `2` as boolean?

Reply:
You are right. I change it to 
```
if (token.equals("TRUE") || token.equals("FALSE") ||token.equals("1")||token.equals("0") ) {
                    return AttributeType.BOOLEAN;
                } else {
                    return AttributeType.STRING;
                }
```
The reason I try to manually rule it is because `Boolean.parse` will parse all String into `True`.

Reply:
Are tokens capitalized or we are not matching for lower cases? I suggest we also match for lower cases